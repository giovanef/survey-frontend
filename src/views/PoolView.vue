<template>
  <div class="pools">
    <template v-if="poolLoading">
      <loading></loading>
    </template>
    <template v-else-if="poolError">
      <error :error="poolError"></error>
    </template>
    <template v-else>
      <pools-view :pool="pool"></pools-view>
    </template>
  </div>
</template>

<script>
import api from '@/services/api';

import Loading from '@/components/ui/Loading.vue';
import Error from '@/components/ui/Error.vue';
import PoolsView from '@/components/features/PoolsView.vue';

export default {
  data() {
    return {
      pool: null,
      poolLoading: true,
      poolError: false,
    };
  },
  mounted() {
    api
      .get(`/pools/${this.$route.params.id}`)
      .then((response) => {
        this.pool = response.data;
        this.poolLoading = false;
      })
      .catch((error) => {
        this.poolError = error;
        this.poolLoading = false;
      });
  },
  components: {
    Loading,
    Error,
    PoolsView,
  },
};
</script>
