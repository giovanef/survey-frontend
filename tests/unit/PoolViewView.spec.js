import { shallowMount } from '@vue/test-utils';

import PoolView from '@/views/PoolView.vue';

describe('PoolViewView', () => {
  const $route = {
    params: {
      id: 1,
    },
  };

  const build = () => {
    const wrapper = shallowMount(PoolView, {
      mocks: {
        $route,
      },
    });

    return {
      wrapper,
    };
  };

  beforeEach(() => {});

  it('renderiza o componente', () => {
    const { wrapper } = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
