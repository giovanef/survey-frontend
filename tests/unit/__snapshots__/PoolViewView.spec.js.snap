// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolViewView renderiza o componente 1`] = `
<div>
  <breadcrumbs-stub items="[object Object],[object Object]"></breadcrumbs-stub>
  <div class="pools">
    <loading-stub></loading-stub>
  </div>
</div>
`;
